packages/pyright-internal/src/analyzer/binder.ts

```diff
@@ -2952,11 +2959,21 @@ export class Binder extends ParseTreeWalker {
                     }
                 }
 
-                if (isPrivateOrProtectedName(name)) {
-                    if (this._fileInfo.isStubFile || isPrivateName(name)) {
-                        symbol.setIsExternallyHidden();
-                    } else if (this._fileInfo.isInPyTypedPackage && this._currentScope.type === ScopeType.Module) {
-                        this._potentialPrivateSymbols.set(name, symbol);
+                if (this._currentScope.type === ScopeType.Module || this._currentScope.type === ScopeType.Builtin) {
+                    if (isPrivateOrProtectedName(name)) {
+                        if (isPrivateName(name)) {
+                            // Private names are obscured, so they are always externally hidden.
+                            symbol.setIsExternallyHidden();
+                        } else if (this._fileInfo.isStubFile || this._fileInfo.isInPyTypedPackage) {
+                            if (this._currentScope.type === ScopeType.Builtin) {
+                                // Don't include private-named symbols in the builtin scope.
+                                symbol.setIsExternallyHidden();
+                            } else {
+                                this._potentialPrivateSymbols.set(name, symbol);
```

Comment:
now symbol in stub can affected by __ all __ when it was always hidden before.

Reply:
Yes, this is intended. See my comment above.